lib/datadog/core/telemetry/logging.rb

```diff
@@ -2,16 +2,51 @@
 
 require_relative 'event'
 
+require 'pathname'
+
 module Datadog
   module Core
     module Telemetry
-      # Logging interface for sending telemetry logs.
+      # === INTRENAL USAGE ONLY ===
+      #
+      # Logging interface for sending telemetry logs... so we can fix them.
+      #
+      # For developer using this module:
+      # - MUST NOT provide any sensitive information (PII)
+      # - SHOULD reduce the data cardinality for batching/aggregation
+      #
+      # Before using it, ask yourself:
+      # - Do we need to know about this (ie. internal error or client error)?
+      # - How severe/critical is this error? (ie. error, warning, fatal)
+      # - What information needed to make it actionable?
       #
-      # Reporting internal error so that we can fix them.
-      # IMPORTANT: Make sure to not log any sensitive information.
       module Logging
         extend self
 
+        # Extract datadog stack trace from the exception
+        module DatadogStackTrace
+          GEM_ROOT = Pathname.new("#{__dir__}/../../../..").cleanpath.to_s
+
+          def self.from(exception)
+            backtrace = exception.backtrace
+
+            return unless backtrace
+            return if backtrace.empty?
+
+            stack_trace = +''
+            backtrace.each do |line|
+              stack_trace << if line.start_with?(GEM_ROOT)
+                               line[GEM_ROOT.length..-1] || ''
+                             else
+                               'REDACTED'
+                             end
+              stack_trace << ','
+            end
+
+            stack_trace.chomp(',')
+          end
+        end
```

Comment:
It occurred to me that if in the future we want to start including stack traces from other datadog libraries (datadog-ci, dogstatsd-ruby), that we don't have any "namespacing" on the current stacks.

E.g. our current stack traces look like

```
/lib/some/file.rb: ...
```

which is fine if we're only allowing files from dd-trace-rb in the stacktrace.

But if we later allow other datadog libraries, how will this look?

```
/lib/some/file.rb: ... # is this file from dd-trace-rb? or not?
/lib/some/other_file.rb: ... # is this file from dd-trace-rb? or not?
```

TL;DR should we prefix the gem name in the stacks we generate?

e.g.

```
datadog/lib/some/file.rb
datadog-ci/lib/some/other_file.rb
dogstatsd-ruby/lib/yet_another/file.rb
```

This is not a problem on this PR, but does mean that if we go as-is, later we'll "invalidate" our stacks if we need to change, so perhaps it's worth considering already prepending `datadog` on this PR?

Reply:
As far as I know, we might need to able to accept more roots to filter, but the our file system convention provides namespaces under `/lib/datadog/..`. 

For example: 
- `/lib/datadog/tracing`
- `/lib/datadog/profiling`